### ISSUE TITLE
Rate Limiter

### ISSUE THREAD

**Issue by bennett000** [darrenmce/oauth2-server#7]
looks like the spirit of this package is to be a full on server (as opposed to a consumable express module) in that case a rate limiter is probably a good idea for production.

There are a few rate limiter packages out there but rate limiting comes with another issue in that you need some sort of fingerprinting and if you get the fingerprinting wrong that's a ripe opportunity for a DOS...

**Comment by darrenmce** [comment]
https://github.com/animir/node-rate-limiter-flexible looks promising, will implement a spike

**Comment by bennett000** [comment]
I love that this is still active.

**Comment by darrenmce** [comment]
was just looking at this again for the first time in a while - man it was fun
/wave @bennett000

**Comment by bennett000** [comment]
@darrenmce i think of it often

I have been spreading your git-gui skills across the pond 😆

**Comment by darrenmce** [comment]
@bennett000 haha amazing - love to hear it!! hopefully the artisan side of this stuff doesn't go away

**Comment by bennett000** [comment]
@darrenmce I don't think it will.  I like McDonald's just fine but I still have an appetite for the Michelin star places
